README.extentions

```diff
@@ -0,0 +1,205 @@
+Escaping of References and Inventory Queries
+============================================
+
+Reference and inventory queries can be escaped to produce literal strings, for example:
+
+parameters:   
+  colour: Blue
+  unescaped: The colour is ${colour}
+  escaped: The colour is \${colour}  
+  double_escaped: The colour is \\${colour}
+
+
+This would produce:           
+
+parameters:   
+  colour: Blue
+  unescaped: The colour is Blue   
+  escaped: The colour is ${colour}   
+  double_escaped: The colour is \Blue
+
+
+
+Merging Referenced Lists and Dictionaries
+=========================================
+
+Referenced lists or dicts can now be merged:
+
+# nodes/test.yml
+classes: 
+  - test1
+  - test2  
+parameters:
+  one:  
+    a: 1
+    b: 2
+  two:  
+    c: 3
+    d: 4
+  three:
+    e: 5
+
+# classes/test1.yml
+parameters:    
+  three: ${one}
+
+# classes/test2.yml
+parameters:    
+  three: ${two}
+
+running reclass.py --nodeinfo node1 then gives:
+
+parameters:
+  one:
+    a: 1
+    b: 2
+  three:
+    a: 1
+    b: 2
+    c: 3
+    d: 4
+    e: 5
+  two:
+    c: 3
+    d: 4
+
+This first sets the parameter three to the value of parameter one (class test1) then merges parameter two into
+parameter three (class test2) and finally merges the parameter three definition given in the node definition into
+
+
+
+Nested References
```

Comment:
lgtm, 
just interested what will be reported as missing key?

Reply:
Given:
```yaml
# class /srv/salt/env/pre-prod/classes/role/server/dns/init.yml
# (node vaars-09)
parameters:
  lookup:
    a: 1
    b: 2
  lookup_key: c
  test: ${lookup:${lookup_key}}
```
The error reported is:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/reclass/values/refitem.py", line 53, in _resolve
    return path.get_value(context)
  File "/usr/lib/python2.7/site-packages/reclass/utils/dictpath.py", line 128, in get_value
    return self._get_innermost_container(base)[self._get_key()]
KeyError: 'c'

-> vaars-09.nikhef.nl
   Cannot resolve ${lookup:c}, at test, in yaml_fs:///srv/salt/env/pre-prod/classes/role/server/dns/init.yml